File: backend/app/schemas/vaga.py
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel


class VagaResponse(BaseModel):
    id: int
    titulo: str
    area: Optional[str] = None
    nivel: Optional[str] = None
    descricao: Optional[str] = None
    status_vaga: Optional[str] = None
    empresa: Optional[str] = None
    tipo_contrato: Optional[str] = None
    modelo_trabalho: Optional[str] = None
    localidade_cidade: Optional[str] = None
    localidade_estado: Optional[str] = None
    salario: Optional[Decimal] = None
    salario_periodicidade: Optional[str] = None
    moeda: Optional[str] = None

File: frontend/types/vaga.ts
export type Nivel = 'Junior' | 'Pleno' | 'Senior' | 'Geral';

export type FonteVaga = 'interna' | 'adzuna' | 'remotive' | 'arbeitnow' | string;

export type TipoContrato = 'CLT' | 'PJ' | 'ESTAGIO' | 'JOVEM_APRENDIZ' | 'CORPORATE';
export type ModeloTrabalho = 'PRESENCIAL' | 'HIBRIDO' | 'ONLINE';

export interface Vaga {
  id: number | string;
  titulo: string;
  area?: string | null;
  nivel?: Nivel | string | null;
  modeloTrabalho?: 'Presencial' | 'Hibrido' | 'Remoto' | string | null;
  localidade?: string | null;
  descricao?: string | null;
  status_vaga?: string | null;
  fonte?: FonteVaga;
  urlOrigem?: string | null;
  empresa?: string | null;
  tipoContrato?: TipoContrato | string | null;
  salario?: number | null;
  salarioPeriodicidade?: 'MENSAL' | 'HORA' | 'ANUAL' | string | null;
  moeda?: string | null;
}
